### PR TITLE
Port #52305 to master

### PR DESCRIPTION
**Pull request by xeacott**
### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dwoz** [comment]
@xeacott Test failures look legit

File: salt/fileclient.py [hunk 1]
```diff
@@ -46,8 +47,6 @@
 )
 from salt.utils.openstack.swift import SaltSwift
 
-# pylint: enable=no-name-in-module,import-error
```
Comment (dwoz) [inline salt/fileclient.py:49]: I'm not down with removing these for the whole file.

File: salt/fileclient.py [hunk 2]
```diff
@@ -551,7 +571,7 @@ def s3_opt(key, default=None):
                     https_enable=s3_opt("https_enable", True),
                 )
                 return dest
-            except Exception as exc:  # pylint: disable=broad-except
```
Comment (dwoz) [inline salt/fileclient.py:554]: This shouldn't be changing unless we are actually going to put a more applicable exception type here (which would be the preference).

**Review by dwoz — CHANGES_REQUESTED** [review]

**Comment by xeacott** [comment]
Thanks for reviewing, I'll add them back. This PR was very troublesome not sure how all the extra changes weren't caught by the linter...

**Review by cmcmarrow — APPROVED** [review]

**Pull request merged**